scripts/chopsticks-test.ts

```diff
@@ -0,0 +1,139 @@
+    let hex = "0x" + [...Array(64)].map(() => Math.floor(Math.random() * 16).toString(16)).join('');
+
+    const keyring = new Keyring();
+    return keyring.encodeAddress(hex, ss58Prefix);
+}
+
+async function checkTransfer(fromChain: ChainName, toChain: ChainName, token: string, bridge: Bridge) {
+    // use a fresh address each time to test the worst-case scenario with the ED
+    const originAdapter = bridge.findAdapter(fromChain);
+    const destAdapter = bridge.findAdapter(toChain);
+
+    const destAddress = getRandomAddress(destAdapter.getSS58Prefix());
+    
+    let min = await firstValueFrom(originAdapter.subscribeMinInput(token, toChain));
+    // don't use console.log because I don't want newline here - I want the OK/FAIL to be added on the same line
+    process.stdout.write(`Transferring ${min} ${token} from ${fromChain} to ${toChain}.${destAddress}... `);
+
+    let expectedDiff = FN.fromInner("1", min.getPrecision());
+
+    const tx = originAdapter.createTx({
+      to: toChain,
+      token,
+      amount: min,
+      address: destAddress,
+      signer: destAddress, // doesn't matter in chopsticks
+    });
+
+    let cfg = {
+        token: token,
+        address: destAddress,
+        amount: expectedDiff,
+        timeout: 15*1000, // 15 sec timeout
+    };
+    let changes = destAdapter.subscribeBalanceChanged(cfg);
+    await submitTx(tx as SubmittableExtrinsic<"rxjs", ISubmittableResult>);
+    let q = changes.pipe(takeWhile((x) => x == BalanceChangedStatus.CHECKING, true));
+
+    let result = await lastValueFrom(q);
+    let success = result == BalanceChangedStatus.SUCCESS;
+    if (success) {
+        console.log("OK");
+    } else {
+        console.log("FAIL");
+    }
+    return success;
+}
+
+async function main(): Promise<void> {
+
+    const availableAdapters: Record<string, BaseCrossChainAdapter> = {
+        kusama: new KusamaAdapter(),
+        statemine: new StatemineAdapter(),
+        kintsugi: new KintsugiAdapter(),
+        karura: new KaruraAdapter(),
+        heiko: new HeikoAdapter(),
+    };
+
+    const bridge = new Bridge({
+        adapters: Object.values(availableAdapters),
+    });
+    const chains = Object.keys(availableAdapters) as ChainName[];
+
+    const provider = new ApiProvider("mainnet"); // we overwrite endpoints, so not really mainnet
+    const endpoints = {
+        kusama: ['ws://127.0.0.1:8004'],
+        kintsugi: ['ws://127.0.0.1:8000'],
+        statemine: ['ws://127.0.0.1:8001'],
+        karura: ['ws://127.0.0.1:8002'],
+        heiko: ['ws://127.0.0.1:8003'],
+    };
+
+    // connect all adapters
+    // const connected = 
+    await lastValueFrom(
+        provider.connectFromChain(chains, endpoints)
+    );
+    // and set apiProvider for each adapter
+    await Promise.all(
+        chains.map((chain) =>
+            availableAdapters[chain].setApi(provider.getApi(chain))
+        )
+    );
+
+    let testcases = [
+        ["karura", "KBTC"],
```

Comment:
Maybe we can use the bridge/adapter logic to determine which routes exist. That can wait of course if we're more interested in getting some tests out asap.

Reply:
Could be nice indeed, but there's also something to say for having an explicit list in the code of what is tokens are tested

Reply:
Good point. I like that.

My concern is that we might forget to add that "latest addition" and then run into issues down the line.
Perhaps we can check the static list against the actually implemented we find as we navigate the adapters' routes?
Again: I'm fine with adding that as a todo/issue to be addressed outside of this PR.

Reply:
Let's leave for future pr

Reply:
https://github.com/interlay/bridge/issues/64